include/irr/asset/IAsset.h

```diff
@@ -194,6 +200,8 @@ class IAsset : virtual public core::IReferenceCounted
 			return true;
 		}
 
+		virtual bool canBeRestoredFrom_recurseDAG(const IAsset* _other) const = 0;
```

Comment:
make it protected

Reply:
cant because an asset has to be able to call it on its dependencies

Reply:
but asset and its dependencies both inherit from `IAsset` ?

Reply:
doesnt matter

Reply:
ok there's a simple workaround for that, move `canBeRestoredFrom_recurseDAG` to protected and also provide this as a workaround:
```
inline bool canBeRestoredFrom_recurseDAG_call(const IAsset* _other) const
{
    return canBeRestoredFrom_recurseDAG(_other);
}
```
then because `canBeRestoredFrom_recurseDAG_call` is not virtual derived classes should be able to call it, correcT?

Reply:
> then because canBeRestoredFrom_recurseDAG_call is not virtual derived classes should be able to call it, correcT?

no, why?
hm, i think workaround would be put this into IAsset
```
inline static bool IAsset::canBeRestoredFrom_recurseDAG_call(const IAsset* _this_child, const IAsset* _other_child)
{
    return _this_child->canBeRestoredFrom_recurseDAG(_other_child);
}
```
and then call in derived classes, i.e. in case of DS:
`canBeRestoredFrom_recurseDAG_call(getLayout(), other->getLayout())`

same for `restoreFromDummy_impl`

Reply:
im doing this